WordPress/src/main/java/org/wordpress/android/ui/people/WPEditTextWithChipsOutlined.kt

```diff
@@ -0,0 +1,729 @@
+
+        helperTextState = savedState.labelState
+
+        val chipsData = savedState.chipsData
+
+        for (chip: ChipData in chipsData) {
+            chipify(chip.text, ItemValidationState.stateFromColor(context, chip.color))
+        }
+
+        styleView(savedState.isFocused, savedState.hasText, false)
+    }
+
+    private data class ChipData(val text: String, val color: Int) : Parcelable {
+        constructor(parcel: Parcel) : this(
+                parcel.readString()!!,
+                parcel.readInt()
+        )
+
+        override fun writeToParcel(parcel: Parcel, flags: Int) {
+            parcel.writeString(text)
+            parcel.writeInt(color)
+        }
+
+        override fun describeContents(): Int {
+            return 0
+        }
+
+        companion object CREATOR : Creator<ChipData> {
+            override fun createFromParcel(parcel: Parcel): ChipData {
+                return ChipData(parcel)
+            }
+
+            override fun newArray(size: Int): Array<ChipData?> {
+                return arrayOfNulls(size)
+            }
+        }
+    }
+
+    private class SavedState : BaseSavedState {
+        var labelState = HelperTextState.HINT_VISIBLE
+        var isFocused = false
+        var hasText = false
+        var chipsData = mutableListOf<ChipData>()
+
+        constructor(superState: Parcelable?) : super(superState) {}
+        private constructor(`in`: Parcel) : super(`in`) {
+            this.labelState = HelperTextState.values()[`in`.readInt()]
+            this.isFocused = `in`.readInt() == 1
+            this.hasText = `in`.readInt() == 1
+            val arrayLen = `in`.readInt()
+            if (arrayLen > 0) {
+                val chipsDataArray = arrayOfNulls<ChipData>(arrayLen)
+                `in`.readTypedArray(chipsDataArray, ChipData.CREATOR)
+                this.chipsData = chipsDataArray.filterNotNull().toMutableList()
+            }
+        }
+
+        override fun writeToParcel(out: Parcel, flags: Int) {
+            super.writeToParcel(out, flags)
+            out.writeInt(labelState.ordinal)
+            out.writeInt(if (this.isFocused) 1 else 0)
+            out.writeInt(if (this.hasText) 1 else 0)
+            if (chipsData.size == 0) {
+                out.writeInt(0)
+            } else {
+                out.writeInt(chipsData.size)
+                out.writeTypedArray(chipsData.toTypedArray(), flags)
+            }
+        }
+
+        companion object {
+            @JvmField
+            val CREATOR: Parcelable.Creator<SavedState> = object : Parcelable.Creator<SavedState> {
+                override fun createFromParcel(`in`: Parcel): SavedState {
+                    return SavedState(`in`)
+                }
+
+                override fun newArray(size: Int): Array<SavedState?> {
+                    return arrayOfNulls(size)
+                }
+            }
+        }
+    }
+
+    companion object {
+        private const val MAX_NUMBER_OF_CHIPS_DEFAULT = 10
```

Comment:
ahah, that was the original intention 😄 . I then went to something like: if you don't set the wpMaxNumberChips then no limits in number of chips (0 is used); use the set limit otherwise. I removed the unused constant if you agree 🙇